product-service/src/handlers/createProduct/index.ts

```diff
@@ -0,0 +1,96 @@
+import { APIGatewayEvent } from 'aws-lambda';
+import { Pool } from 'pg';
+import { v4 } from 'uuid';
+
+import { logger } from 'src/utils/logger';
+import { dbOptions } from '../../utils/dbConnection';
+import {
+  buildResponseFailure,
+  buildSuccessResponse,
+  LambdaResponseSerialized,
+} from '../../utils/responses';
+import { validation } from '../../utils/validation';
+
+const pool = new Pool(dbOptions);
+
+export const createProduct = async (
+  event: APIGatewayEvent
+): Promise<LambdaResponseSerialized> => {
+  if (!event.body) {
+    logger.error('No product data provided');
+    return buildResponseFailure(400, 'Bad request: event.body is empty');
+  }
+
+  const { title, description, price, imgUrl, count } = JSON.parse(event.body);
+  logger.info(
+    `Creating product: title: ${title}, description: ${description}, price: ${price}, imgUrl: ${imgUrl}, count: ${count}`
+  );
+
+  const data = validation(title, description, price, imgUrl, count);
+  if (data instanceof Error) {
+    logger.error(`Error validating product: ${data.message}`);
+    return buildResponseFailure(400, `Validation error: ${data.message}`);
+  }
+
+  const id = v4();
+  const client = await pool.connect();
+  logger.info('Connected to database');
+
+  await client.connect().catch((err) => {
+    logger.error(`Error connecting to database: ${err.message}`);
+    return buildResponseFailure(500, err);
+  });
+
+  try {
+    await client.query('BEGIN').then(() => logger.info('Transaction started'));
+
+    await client
+      .query(
+        `
+    	insert into products (id, title, description, price, imgUrl)
+    	values (
+        '${id}',
```

Comment:
"Avoid string concatenating parameters into the query text directly. This can (and often does) lead to sql injection vulnerabilities. Node-postgres supports parameterized queries"
https://node-postgres.com/features/queries

For example:
```
const text = 'INSERT INTO users(name, email) VALUES($1, $2) RETURNING *'
const values = ['brianc', 'brian.m.carlson@gmail.com']

client.query(text, values)
```

Reply:
Thanks. great note. I even thought about it